src/test/java/core/basesyntax/service/handlers/AddOperationStrategyTest.java

```diff
@@ -0,0 +1,55 @@
+package core.basesyntax.service.handlers;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.FruitDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.dto.FruitRecordDto;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.Operation;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class AddOperationStrategyTest {
+    private static FruitDao fruitDao;
+    private static RecordHandler addOperation;
+    private static Fruit fruit;
+    private static FruitRecordDto firstFruitRecordDto;
+    private static FruitRecordDto secondFruitRecordDto;
+    private static final int FIRST_POSITIVE = 345;
+    private static final int SECOND_POSITIVE = 275;
+
+    @Rule
+    public ExpectedException expectedEx = ExpectedException.none();
+
+    @BeforeClass
+    public static void setUp() {
+        fruitDao = new FruitDaoImpl();
+        addOperation = new AddOperationStrategy(fruitDao);
+        fruit = new Fruit("Grapes");
+    }
+
+    @Test
+    public void check_addOperation_OK() {
+        firstFruitRecordDto =
+                new FruitRecordDto(Operation.getOperationByLetter("b"), fruit, FIRST_POSITIVE);
+        secondFruitRecordDto =
+                new FruitRecordDto(Operation.getOperationByLetter("b"), fruit, SECOND_POSITIVE);
+        addOperation.applyAction(firstFruitRecordDto);
+        addOperation.applyAction(secondFruitRecordDto);
+        int expected = 620;
+        Assert.assertEquals(expected, Storage.fruits.get(fruit).intValue());
```

Comment:
Is the BALANCE operation should add? I think this operation must return quantity at the end of the line...

Reply:
yes it should add

Reply:
I think this operation initial for the morning for each fruit and shows how much fruit in the warehouse :)

Reply:
so what should i do with those numbers, leave them be?
i need to add them to my Storage so I can work with them

Reply:
ok, that's your opinion